### PR TITLE
Change to use AnyCPU to address Mono issue

### LINKED ISSUE #1247: CakeExecuteScript access denied errors on mono/mac
### What You Are Seeing?
Getting an access denied error
```
Error: ApplicationName='/Users/redth/Desktop/breakcake/tools/Cake/Cake.exe', CommandLine='"/Users/redth/Desktop/breakcake/two.cake"', CurrentDirectory='/Users/redth/Desktop/breakcake', Native error= Access denied
```

### What is Expected?
No errors, secondary script should execute successfully

### What version of Cake are you using?
0.16

### Are you running on a 32 or 64 bit system?
x64

### What environment are you running on?  Windows? Linux? Mac?
Mac 10.11, Mac 10.12

### Are you running on a CI Server?  If so, which one?
N/A

### How Did You Get This To Happen? (Steps to Reproduce)

build.cake:
```
Information ("Running: {0}", "build.cake");

// Run the script from the subfolder
CakeExecuteScript ("two.cake");
```

two.cake:
```
Information ("Running: {0}", "two.cake");
```

### Output Log

~~~sh
Redth:breakcake redth$ cake --verbosity=diagnostic
Module directory does not exist.
Analyzing build script...
Analyzing /Users/redth/Desktop/breakcake/build.cake...
Processing build script...
Creating script session...
Adding reference to System.Data.dll...
Adding reference to System.Xml.dll...
Adding reference to System.Xml.Linq.dll...
Adding reference to Cake.Core.dll...
Adding reference to Cake.Common.dll...
Adding reference to Cake.exe...
Importing namespace Cake.Common...
Importing namespace Cake.Common.Build...
Importing namespace Cake.Common.Build.AppVeyor...
Importing namespace Cake.Common.Build.AppVeyor.Data...
Importing namespace Cake.Common.Build.BitbucketPipelines...
Importing namespace Cake.Common.Build.BitbucketPipelines.Data...
Importing namespace Cake.Common.Build.Bitrise...
Importing namespace Cake.Common.Build.Bitrise.Data...
Importing namespace Cake.Common.Build.ContinuaCI...
Importing namespace Cake.Common.Build.ContinuaCI.Data...
Importing namespace Cake.Common.Build.Jenkins...
Importing namespace Cake.Common.Build.Jenkins.Data...
Importing namespace Cake.Common.Build.TravisCI...
Importing namespace Cake.Common.Build.TravisCI.Data...
Importing namespace Cake.Common.Diagnostics...
Importing namespace Cake.Common.IO...
Importing namespace Cake.Common.IO.Paths...
Importing namespace Cake.Common.Net...
Importing namespace Cake.Common.Security...
Importing namespace Cake.Common.Solution...
Importing namespace Cake.Common.Solution.Project...
Importing namespace Cake.Common.Solution.Project.Properties...
Importing namespace Cake.Common.Solution.Project.XmlDoc...
Importing namespace Cake.Common.Text...
Importing namespace Cake.Common.Tools...
Importing namespace Cake.Common.Tools.Cake...
Importing namespace Cake.Common.Tools.Chocolatey...
Importing namespace Cake.Common.Tools.Chocolatey.ApiKey...
Importing namespace Cake.Common.Tools.Chocolatey.Config...
Importing namespace Cake.Common.Tools.Chocolatey.Features...
Importing namespace Cake.Common.Tools.Chocolatey.Install...
Importing namespace Cake.Common.Tools.Chocolatey.Pack...
Importing namespace Cake.Common.Tools.Chocolatey.Pin...
Importing namespace Cake.Common.Tools.Chocolatey.Push...
Importing namespace Cake.Common.Tools.Chocolatey.Sources...
Importing namespace Cake.Common.Tools.Chocolatey.Upgrade...
Importing namespace Cake.Common.Tools.DNU...
Importing namespace Cake.Common.Tools.DNU.Build...
Importing namespace Cake.Common.Tools.DNU.Pack...
Importing namespace Cake.Common.Tools.DNU.Restore...
Importing namespace Cake.Common.Tools.DotCover...
Importing namespace Cake.Common.Tools.DotCover.Analyse...
Importing namespace Cake.Common.Tools.DotCover.Cover...
Importing namespace Cake.Common.Tools.DotNetCore...
Importing namespace Cake.Common.Tools.DotNetCore.Build...
Importing namespace Cake.Common.Tools.DotNetCore.Execute...
Importing namespace Cake.Common.Tools.DotNetCore.Pack...
Importing namespace Cake.Common.Tools.DotNetCore.Publish...
Importing namespace Cake.Common.Tools.DotNetCore.Restore...
Importing namespace Cake.Common.Tools.DotNetCore.Run...
Importing namespace Cake.Common.Tools.DotNetCore.Test...
Importing namespace Cake.Common.Tools.DupFinder...
Importing namespace Cake.Common.Tools.Fixie...
Importing namespace Cake.Common.Tools.GitLink...
Importing namespace Cake.Common.Tools.GitReleaseManager...
Importing namespace Cake.Common.Tools.GitReleaseManager.AddAssets...
Importing namespace Cake.Common.Tools.GitReleaseManager.Close...
Importing namespace Cake.Common.Tools.GitReleaseManager.Create...
Importing namespace Cake.Common.Tools.GitReleaseManager.Export...
Importing namespace Cake.Common.Tools.GitReleaseManager.Publish...
Importing namespace Cake.Common.Tools.GitReleaseNotes...
Importing namespace Cake.Common.Tools.GitVersion...
Importing namespace Cake.Common.Tools.ILMerge...
Importing namespace Cake.Common.Tools.ILRepack...
Importing namespace Cake.Common.Tools.InspectCode...
Importing namespace Cake.Common.Tools.MSBuild...
Importing namespace Cake.Common.Tools.MSTest...
Importing namespace Cake.Common.Tools.NSIS...
Importing namespace Cake.Common.Tools.NuGet...
Importing namespace Cake.Common.Tools.NuGet.Install...
Importing namespace Cake.Common.Tools.NuGet.Pack...
Importing namespace Cake.Common.Tools.NuGet.Push...
Importing namespace Cake.Common.Tools.NuGet.Restore...
Importing namespace Cake.Common.Tools.NuGet.SetApiKey...
Importing namespace Cake.Common.Tools.NuGet.SetProxy...
Importing namespace Cake.Common.Tools.NuGet.Sources...
Importing namespace Cake.Common.Tools.NuGet.Update...
Importing namespace Cake.Common.Tools.NUnit...
Importing namespace Cake.Common.Tools.OctopusDeploy...
Importing namespace Cake.Common.Tools.OpenCover...
Importing namespace Cake.Common.Tools.ReportGenerator...
Importing namespace Cake.Common.Tools.ReportUnit...
Importing namespace Cake.Common.Tools.Roundhouse...
Importing namespace Cake.Common.Tools.SignTool...
Importing namespace Cake.Common.Tools.SpecFlow...
Importing namespace Cake.Common.Tools.SpecFlow.StepDefinitionReport...
Importing namespace Cake.Common.Tools.SpecFlow.TestExecutionReport...
Importing namespace Cake.Common.Tools.TextTransform...
Importing namespace Cake.Common.Tools.VSTest...
Importing namespace Cake.Common.Tools.WiX...
Importing namespace Cake.Common.Tools.WiX.Heat...
Importing namespace Cake.Common.Tools.XBuild...
Importing namespace Cake.Common.Tools.XUnit...
Importing namespace Cake.Common.Xml...
Importing namespace Cake.Core...
Importing namespace Cake.Core.Diagnostics...
Importing namespace Cake.Core.IO...
Importing namespace Cake.Core.Scripting...
Importing namespace System...
Importing namespace System.Collections.Generic...
Importing namespace System.IO...
Importing namespace System.Linq...
Importing namespace System.Text...
Importing namespace System.Threading.Tasks...
Compiling build script...
Running: build.cake
Executing: /Users/redth/Desktop/breakcake/tools/Cake/Cake.exe "/Users/redth/Desktop/breakcake/two.cake"
Error: System.ComponentModel.Win32Exception (0x80004005): ApplicationName='/Users/redth/Desktop/breakcake/tools/Cake/Cake.exe', CommandLine='"/Users/redth/Desktop/breakcake/two.cake"', CurrentDirectory='/Users/redth/Desktop/breakcake', Native error= Access denied
  at System.Diagnostics.Process.StartWithCreateProcess (System.Diagnostics.ProcessStartInfo startInfo) [0x001ee] in <affe4060066c42de8cdd6027cdb92b56>:0
~~~

### PR DESCRIPTION
**Pull request by devlead**
This fixes #1247, because issue in .NET Core tooling AnyCPU needs to be specified

### CODE REVIEW

**Comment by devlead** [comment]
So @cake-build/team & @Redth 
Fetched the AppVeyor artifacts ( https://ci.appveyor.com/api/buildjobs/jc4h8asmrbm4c9je/artifacts/Cake-bin-net45-v0.17.0-PullRequest1266.zip )  all integration tests pass 👍 
image

1. I cloned out this PR to an Ubuntu machine
2. Unzipped artifact to tools\Cake
3. Ran `mono tools/Cake/Cake.exe ./build.cake --verbosity=Diagnostic --target=Cake.Common.Tools.Cake.CakeAliases`
4. Success

This PR removes the mono framework prefixing in integration tests which is confirmed by the diagnostic log

```bash
========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/resources/Cake.Common/Tools/Cake/build.cake"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript

========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.Ok
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.Ok
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/resources/Cake.Common/Tools/Cake/test.cake" -ok="yes"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.Ok

========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.NotOk
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.NotOk
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/resources/Cake.Common/Tools/Cake/test.cake" -ok="no"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteScript.Settings.NotOk

========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/c497a5df-be2d-4a91-8373-47d3ce82ccb9.cake"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression

========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.Ok
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.Ok
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/0d8dbaa3-eded-4baf-ac74-a371ad0742c2.cake" -ok="yes"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.Ok

========================================
Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.NotOk
========================================
Executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.NotOk
Executing: /home/matkar/monocakescriptissue/cake/tests/integration/tools/Cake/Cake.exe "/home/matkar/monocakescriptissue/cake/tests/integration/d86579d2-4618-4a53-ac79-c76cd1c7cdd1.cake" -ok="no"
Finished executing task: Cake.Common.Tools.Cake.CakeAliases.CakeExecuteExpression.Settings.NotOk
```

To confirm fix i downloaded v16.1 artifacts form last integration test ( https://ci.appveyor.com/api/buildjobs/sbpo9nvvjjy2pgmr/artifacts/Cake-bin-net45-v0.16.1.zip ) and did same procedure as above and it as expected failed:
image

So this PR is ready for review.

File: src/Cake/project.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ﻿{
-  "version": "0.16.1-*",
+  "version": "0.16.2-*",
```
Comment (gep13) [inline src/Cake/project.json]: @devlead doesn't this change need to happen in a number of files?  Not just the top level one?

Comment (devlead) [inline src/Cake/project.json]: Think that'll be patched by build.cake&GitVersion during build. As AppVeyor PR artifacts all  v0.17.0 both nuget & assemblies, hopefully hotfix branch will get correct versioning once PR merged :)

Comment (gep13) [inline src/Cake/project.json]: @devlead the reason that I ask is that in the last PR you did for 0.16.1, you updated all the files 😄

Comment (devlead) [inline src/Cake/project.json]: Done.

File: src/Cake/project.json [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   "buildOptions": {
     "emitEntryPoint": true,
     "xmlDoc": true,
+    "platform": "AnyCpu",
```
Comment (patriksvensson) [inline src/Cake/project.json]: I think change should be added to all assemblies.

Comment (devlead) [inline src/Cake/project.json]: Ok, I'll fix.

Comment (devlead) [inline src/Cake/project.json]: Done.

**Pull request merged**